client/src/ui/molecules/search/index.tsx

```diff
@@ -1,62 +1,124 @@
-import React, { Suspense, lazy, useEffect, useState } from "react";
+import React, {
+  Suspense,
+  lazy,
+  useState,
+  useMemo,
+  useRef,
+  useEffect,
+} from "react";
+import { useSearchParams } from "react-router-dom";
 
-import { AUTOCOMPLETE_SEARCH_WIDGET } from "../../../constants";
 import { useLocale } from "../../../hooks";
 
 import "./index.scss";
 
-// The temporary solution... See below around the BasicSearchWidget component.
 import "./basic-search-widget.scss";
-// import { SearchNavigateWidget } from "../../../search";
-const LazySearchNavigateWidget = lazy(() => import("./lazy-search-widget"));
+import {
+  getPlaceholder,
+  SearchProps,
+  useFocusOnSlash,
+} from "../../../search-utils";
+const LazySearchNavigateWidget = lazy(() => import("../../../search"));
+
+function useQueryParamState() {
+  const [searchParams] = useSearchParams();
+  const queryState = searchParams.get("q") || "";
+  const [value, setValue] = useState(queryState);
+
+  // The site-search page might trigger an update to the current
+  // `?q=...` value and if that happens we want to be reflected in the search inputs
+  React.useEffect(() => {
+    setValue(queryState);
+  }, [setValue, queryState]);
+
+  return [value, setValue] as const;
+}
 
 export function Search(props) {
-  const [useAutocompleteSearchWidget, setUseAutocompleteSearchWidget] =
-    useState(false);
+  const [value, setValue] = useQueryParamState();
+  const [isFocused, setIsFocused] = useState(false);
+  const [defaultSelection, setDefaultSelection] = useState([0, 0] as const);
+  const [hadFocus, setHadFocus] = useState(false);
+
+  const searchProps = useMemo(
+    () => ({
+      inputValue: value,
+      onChangeInputValue: (value) => setValue(value),
+      isFocused,
+      onChangeIsFocused: (isFocused) => setIsFocused(isFocused),
+      defaultSelection,
+      onChangeSelection: (selection) => setDefaultSelection(selection),
+    }),
+    [
+      value,
+      setValue,
+      isFocused,
+      setIsFocused,
+      defaultSelection,
+      setDefaultSelection,
+    ]
+  );
+
   useEffect(() => {
-    if (AUTOCOMPLETE_SEARCH_WIDGET) {
-      setUseAutocompleteSearchWidget(true);
+    if (isFocused) {
+      setHadFocus(true);
     }
-  }, []);
+  }, [isFocused, setHadFocus]);
 
   return (
     <div className="header-search">
-      {/* See the code comment next to the <BasicSearchWidget> component */}
-      {useAutocompleteSearchWidget ? (
-        <Suspense fallback={<BasicSearchWidget />}>
-          <LazySearchNavigateWidget {...props} />
+      {hadFocus ? (
+        <Suspense fallback={<BasicSearchWidget {...searchProps} />}>
+          <LazySearchNavigateWidget {...searchProps} {...props} />
         </Suspense>
       ) : (
-        <BasicSearchWidget />
+        <BasicSearchWidget {...searchProps} />
```

Comment:
I don't know how or exactly where but I think it would be nice to kick off on `onMouseOver` *too*. I.e. Focus or hover over. I notice myself, there's often slight delay when I do use the mouse to first hover over the basic input and then click into it. Perhaps it's over-optimization but that's potentially a couple of hundred milliseconds of valuable time which we could fill with fetching the external JS.